### PR TITLE
Fix typescript build errors

### PR DESCRIPTION
**Pull request by gvnn**
Hi,
Really excited about this pull PR on `postgraphql`, but we couldn't build the package, so I have managed to getting it building. I'm not a typescript dev, so I have probably broken every rule of TypeScript development.

Cheers,
G.

### CODE REVIEW

File: src/postgraphql/cli.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { resolve as resolvePath } from 'path'
 import { readFileSync } from 'fs'
 import { createServer } from 'http'
 import chalk = require('chalk')
-import { Command } from 'commander'
+import program = require('commander')
```
Comment (mikeburgh) [inline src/postgraphql/cli.ts]: Not sure about these changes, I did not touch them in the original fork, so I think we might need to leave these as they were

**Comment by mikeburgh** [comment]
Thanks! we are both not Typescript dev's but I can see you fixed up some typing. Not sure about changing that cli.ts file imports as they were like that in the original fork. 
As for the requireGlob change, let me test that to see if it passes the upstream tests, and if so I can accept them.

**Comment by gvnn** [comment]
Yeah I wasn't sure about `cli.ts` because that was the original code, for some reason it wasn't recognising the `@types`. I'll try to see if I can fix it.

**Comment by gvnn** [comment]
I tried to fix it... my local refuses to use the commander import like it was before :(

**Comment by mikeburgh** [comment]
yeah, strangest thing, I cannot build it with it either.. it looks like the main repo is good to accept this if I make some changes, so going to tackle those tomorrow and then update the PR in the main. I will raise the commander issue with them and see what they say

**Comment by gvnn** [comment]
Hey @mikeburgh have you thought on how to inject the context in the custom mutations?

**Comment by gvnn** [comment]
Just a thought... what if we inject pre-made extensions? this should allow injecting manually jwt and others.

**Comment by mikeburgh** [comment]
Yeah I had some feedback from the main repo, about making the schema injections export a function rather than the schema. That way you pass into the function the postgraphSQL object and get access to it in the function, which in turn returns the schema.

I plan on implementing that tomorrow and some tests, and hopefully get the MR accepted in the up stream repo. I need to also confirm you can get at the JWT via the execute context method or the postgraphSQL object.. otherwise we still need a way to do that

I had a thought do do what you did and push in an array, but it wont work from the command line interface, and not sure if the upstream guys want to keep both interfaces the same..

**Comment by gvnn** [comment]
Oh yeah... you are right, oops... I'll let you do your PR and then we'll update our code once it's ready ;)

**Comment by mikeburgh** [comment]
No worries, should get it sorted tomorrow

*truelogic*
AU Direct: 02 8007 6731
US Direct: 714 800 0379

**Comment by gvnn** [comment]
btw... thanks a lot! I had to progress on this because we are a little bit blocked, so I mocked something together.

**Comment by mikeburgh** [comment]
No worries at all, I understand what that's like.. hopefully we get it into the main branch soon and then it's done.. 

I also had an idea about intercepting the schema that is automatically created via the introspection.. so you could have a method say run after one of the store procs (or before it).. I need to test that out though to see if it's practical.. and the reality is it might be neater to just create an injected mutation, have it call the SP rather than trying to intercept it.

**Comment by mikeburgh** [comment]
Hey, so I refactored this to return a function.. if you check out the upstream merge you can see the example now of what an injection file should look like: https://github.com/postgraphql/postgraphql/pull/448 

Still working out the best way to get the jwt data there, role is added to that context argument in the resolve method, but not the rest of the token data.

**Comment by mikeburgh** [comment]
I just exposed jwtClaims to the context argument.. so that's done.. now to just figure out connecting to the DB when it should not

**Comment by gvnn** [comment]
That's great! it's mostly what we need. We have found another solution, we are not using the express http middleware and creating the context manually, this means that before launching the graphql query we inject the express request in the context... it's an hack but kinda works

**Pull request closed without merging**